homeassistant/components/ohme/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""The ohme integration."""
+
+from dataclasses import dataclass
+import logging
+
+from ohme import OhmeApiClient
+
+from homeassistant import core
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONFIG_VERSION, ENTITY_TYPES
+from .coordinator import OhmeAdvancedSettingsCoordinator, OhmeChargeSessionsCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+type OhmeConfigEntry = ConfigEntry[OhmeRuntimeData]
+
+
+@dataclass
+class OhmeRuntimeData:
+    """Store volatile data."""
+
+    client: OhmeApiClient
+    coordinators: list[DataUpdateCoordinator]
+
+
+async def async_update_listener(hass, entry):
+    """Handle options flow credentials update."""
+
+    # Reload this instance
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Ohme from a config entry."""
+
+    client = OhmeApiClient(entry.data["email"], entry.data["password"])
+
+    entry.runtime_data = OhmeRuntimeData(client, [])
```

Comment:
```suggestion
```
should be enough if we assign it after the initial update, since you will have coordinators in any case